docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,562 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
 
   import {symbol1 as alias, symbol2} from "filename";
 
-Paths
------
+.. _path-resolution:
 
-In the above, ``filename`` is always treated as a path with ``/`` as directory separator,
-and ``.`` as the current and ``..`` as the parent directory.  When ``.`` or ``..`` is followed by a character except ``/``,
-it is not considered as the current or the parent directory.
-All path names are treated as absolute paths unless they start with the current ``.`` or the parent directory ``..``.
+Path Resolution
+---------------
 
-To import a file ``filename`` from the same directory as the current file, use ``import "./filename" as symbolName;``.
-If you use ``import "filename" as symbolName;`` instead, a different file could be referenced
-(in a global "include directory").
+The paths used in imports are not filesystem paths.
+In the simplest cases they may end up being used directly to load files from disk but in general
+the final path can be substantially different due to abstractions necessary to ensure reproducible
+builds on different platforms.
 
-It depends on the compiler (see :ref:`import-compiler`) how to actually resolve the paths.
-In general, the directory hierarchy does not need to strictly map onto your local
-filesystem, and the path can also map to resources such as ipfs, http or git.
+.. index:: import key, import path, filesystem path
 
-.. note::
-    Always use relative imports like ``import "./filename.sol";`` and avoid
-    using ``..`` in path specifiers. In the latter case, it is probably better to use
-    global paths and set up remappings as explained below.
+Virtual Filesystem
+~~~~~~~~~~~~~~~~~~
+
+The compiler maintains an internal database (virtual filesystem) where each compiled module is
+assigned a unique *import key* which is an opaque and unstructured identifier.
```

Comment:
I'm not sure if we can say that, since `.` and `..` actually use structure inside these identifiers.

Reply:
In these docs I distinguish between "import paths" (paths you specify in import statement) and "import key" (the ID that ends up assigned to the file in the virtual filesystem). AFAIK `.` and `..` only have special meaning in import paths, not import keys.